File: src/Goods.Domain/LocalAreas/LocalArea.cs
using Goods.Domain.LocalAreas.Enums;

namespace Goods.Domain.LocalAreas;

public class LocalArea(
    Guid id,
    String name,
    AreaType type,
    Int32 population,
    DateOnly establishmentDate,
    Double averageHotelBill,
    Boolean isHeroCity,
    Guid regionId
)
{
    public Guid Id { get; } = id;
    public String Name { get; } = name;
    public AreaType Type { get; } = type;
    public Int32 Population { get; } = population;
    public DateOnly EstablishmentDate { get; } = establishmentDate;
    public Double AverageHotelBill { get; } = averageHotelBill;
    public Boolean IsHeroCity { get; } = isHeroCity;
    public Guid RegionId { get; } = regionId;
}

File: src/Areas.BackOffice/client-app/src/domain/localAreas/localArea.ts
import { Page } from "../../tools/types/page";
import { AreaType } from "./enums/areaType";

export class LocalArea {
  constructor(
    public readonly id: string,
    public readonly name: string,
    public readonly areaType: AreaType,
    public readonly population: number,
    public readonly establishmentDate: Date,
    public readonly averageHotelBill: number,
    public readonly isHeroCity: boolean,
    public readonly regionId: string,
  ) {}
}

export function mapToLocalAreasPage(data: any): Page<LocalArea> {
  return Page.convert(data, mapToLocalArea);
}

export function mapToLocalAreas(data: any[]): LocalArea[] {
  return data.map(mapToLocalArea);
}

export function mapToLocalArea(data: any): LocalArea {
  return new LocalArea(
    data.id,
    data.name,
    data.areaType,
    data.population,
    data.establishmentDate,
    data.averageHotelBill,
    data.isHeroCity,
    data.regionId,
  );
}
